os/kernel/binary_manager/binary_manager_recovery.c

```diff
@@ -137,31 +137,25 @@ static void binary_manager_recover_tcb(struct tcb_s *tcb)
  *	 This function excludes all active threads in a same binary from scheduling.
  *
  ****************************************************************************/
-static int binary_manager_deactivate_binary(pid_t binid)
+static int binary_manager_deactivate_binary(int bin_idx)
 {
 	irqstate_t flags;
-	struct tcb_s *tcb;
-
-	tcb = sched_gettcb(binid);
-	if (tcb == NULL) {
-		return ERROR;
+	struct tcb_s *ptr;
+
```

Comment:
You should check that bin_idx is greater than zero here.
That is, if (bin_idx > 0) {